src/components/diagrams/networkAreaDiagram/network-area-diagram-content.jsx

```diff
@@ -159,15 +188,35 @@ function NetworkAreaDiagramContent(props) {
 
             diagramViewerRef.current = diagramViewer;
         }
-    }, [props.diagramId, props.svgType, props.svg, currentNode, props.loadingState, diagramSizeSetter]);
+    }, [
+        props.diagramId,
+        props.svgType,
+        props.svg,
+        currentNode,
+        props.loadingState,
+        diagramSizeSetter,
+        handleTogglePopover,
+    ]);
 
+    const displayTooltip = () => {
```

Comment:
i think it better for performance to not use a function to render this component